platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/validation/StaticValidators.java

```diff
@@ -63,33 +60,34 @@ public static boolean isValidTimeCreated(final EventImpl event) {
     }
 
     /**
-     * Validates if an event's parent data is correct
+     * Creates a validator that checks if an event's parent data is correct
      *
-     * @param event
-     * 		the event to validate
-     * @return true if the event is valid, false otherwise
+     * @param networkSize the size of the network
+     * @return an event validator that checks if the parent data is valid
      */
-    public static boolean isParentDataValid(final BaseEvent event) {
-        final BaseEventHashedData hashedData = event.getHashedData();
-        final Hash spHash = hashedData.getSelfParentHash();
-        final Hash opHash = hashedData.getOtherParentHash();
-        final boolean hasSpHash = spHash != null;
-        final boolean hasOpHash = opHash != null;
-        final boolean hasSpGen = hashedData.getSelfParentGen() >= GraphGenerations.FIRST_GENERATION;
-        final boolean hasOpGen = hashedData.getOtherParentGen() >= GraphGenerations.FIRST_GENERATION;
+    public static GossipEventValidator isParentDataValid(final int networkSize) {
```

Comment:
would you be opposed to renaming the method to `getParentDataValidator` or similar? The `is` verbiage implies returning a boolean

Reply:
renamed to `buildParentValidator`